### ISSUE TITLE
Restrict what submissions users/clients can view and edit

### ISSUE THREAD

**Issue by crsleeth** [formtools/core#468]
Is it possible to restrict what submissions users/clients can view and edit? For example, make it so users can only view and edit their own submission and not other user's. I'm hoping to create a site where users can maintain a yearly profile about themselves with info like name, age, location, etc. that they can come back to every year to update. Their info needs to be private--other users shouldn't be able to see or edit submissions they don't own. This is for an internal form.

**Comment by benkeen** [comment]
Hey @crsleeth, sorry for not getting back to you sooner. Busy few days...! 

The standard way for this it to use Views. You can configure as many Views as you'd like, each with their own settings. So one View could have readonly fields, for example, another could be editable, but only show a subset of the form fields, or even submissions according to various rules (like only show submissions with particular values in certain fields).

However, the scenario you described isn't solved very easily just yet. Form Tools has the option to add [Client Map filters](https://docs.formtools.org/userdoc/views/client_map_filters/) which limit the submissions that appear in a View by comparing the contents of a particular form field to a value in the client's own user account. To put it a bit clearer, say you have a form field called "user", which has numeric values of 1, 2 and 3. Then you have three client accounts each with those account IDs. By adding a Client 

The trick, however, is actually populating that form field with the appropriate value for a user. The only way to do that right now is through elaborate hacks using the [Submission Pre-Parser](https://docs.formtools.org/modules/submission_pre_parser/) module, to detect what user is currently logged in, then populate that hidden field 

But tell you what, this is a long overdue feature so I'd like to add it in. What I'm thinking of doing is this: in the Edit Form -> Views tab -> General subtab, there's a setting called "Default Values for New Submissions". That lets you do just that - provide default values for specific fields in all new form submissions. If I added _placeholder_ which you could enter in those fields, like `{$account_id}`, *IF* that value is defined (i.e. if the submissions is being created through the Form Tools interface by a logged in user) it would fill in that field with the value when the submission is first added. Then you could just add a simple Client Map filter in your View to limit the results to only contain those submissions that the user him/herself created. THAT view would all be editable, but another view could list all users submissions and be readonly. 

Is this kinda clear? Would this help?

**Comment by crsleeth** [comment]
Hi Ben. I've had a busy few weeks as well so sorry for the late reply. Your response is clear and I would love to see that feature. I see you've added it as a 3.0.13 feature. Do you have a timeline for when 3.0.13 will be released? No rush, I very much appreciate your efforts on this project.

**Comment by benkeen** [comment]
To be honest, I thought it would be out the door already! But I ran into a number of complications with another feature in 3.0.13 (conflict resolution when users simultaneously edit submissions) which are taking some time to resolve. But I'd say this coming weekend would be a good bet.

**Comment by benkeen** [comment]
Hey @crsleeth, alrighty! Released at last. I wrote up this tutorial which is probably the best place to follow how to configure it all: 
https://docs.formtools.org/tutorials/client_accounts_edit_own_data_only/

It'll require Form Tools 3.0.13 - and if you want to use any additional placeholder from the Extended Client Fields module, you'll need to update that to the latest version too (2.1.0). 

All the best!
